askama_shared/src/parser.rs

```diff
@@ -879,24 +897,29 @@ fn block_for<'a>(i: &'a [u8], s: &'a Syntax<'a>) -> IResult<&'a [u8], Node<'a>>
                     cut(tuple((
                         |i| tag_block_start(i, s),
                         opt(tag("-")),
+                        opt(else_block),
                         ws(tag("endfor")),
                         opt(tag("-")),
                     ))),
                 ))),
             ))),
         ))),
     ));
-    let (i, (pws1, _, (var, _, (iter, nws1, _, (block, (_, pws2, _, nws2)))))) = p(i)?;
-    Ok((
-        i,
-        Node::Loop(
-            Ws(pws1.is_some(), nws1.is_some()),
-            var,
-            iter,
-            block,
-            Ws(pws2.is_some(), nws2.is_some()),
-        ),
-    ))
+    let (i, (pws1, _, (var, _, (iter, nws1, _, (block, (_, pws2, else_block, _, nws2)))))) = p(i)?;
+    match else_block {
+        None => {
+            let ws1 = Ws(pws1.is_some(), nws1.is_some());
```

Comment:
Let's hoist the definition of `ws1` out of the `match` expression.

Reply:
Done.